### PR TITLE
Adding a new decorator for data (ISO8601 YYYY-MM-DD format)

### PR DESCRIPTION
**Pull request by adrian-hippo**
The decorator name is aligned with the name from `class-validator` module for now, since it is consistent with the names of all the other decorators. I don't think `class-validator` consistency is a requirement, but it makes it more intuitive for now until we decide underwise.

### CODE REVIEW

File: src/flavors/__tests__/__snapshots__/openapi.test.ts.snap [hunk 1]
```diff
@@ -153,6 +171,7 @@ Object {
           "requiredStringValue",
           "requiredUUIDValue",
           "optionalBooleanValue",
+          "optionalDateStringValue",
```
Comment (adrian-hippo) [inline src/flavors/__tests__/__snapshots__/openapi.test.ts.snap]: TODO: check why optional fields are in the list of required fields.

File: src/flavors/__tests__/__snapshots__/openapi.test.ts.snap [hunk 2]
```diff
@@ -145,21 +162,14 @@ Object {
...
           "requiredBooleanValue",
+          "requiredDateStringValue",
           "requiredDateValue",
           "requiredEnumValue",
           "requiredIntegerValue",
           "requiredNestedValue",
           "requiredNumberValue",
           "requiredStringValue",
           "requiredUUIDValue",
-          "optionalBooleanValue",
```
Comment (adrian-hippo) [inline src/flavors/__tests__/__snapshots__/openapi.test.ts.snap:155]: I found that OpneAPI decorators were not using the optional flag, so included a fix in this PR.

File: src/flavors/__tests__/fixtures.ts [hunk 3]
```diff
@@ -46,6 +47,11 @@ export function createFixtures(
         @IsBoolean()
         requiredBooleanValue!: boolean;
 
+        @IsDate({
+            format: DateFormat.DATE,
```
Comment (jessemyers) [inline src/flavors/__tests__/fixtures.ts]: I have a few thoughts here:
 - We might want separate decorators for `IsDate` and `IsDateTime`.
 - We might want separate decorators for `Date` and date strings.
 - If we choose to keep the format, we might want to have a test that allows passing a string input; I'm not convinced that everyone will want to import and use the enum.

File: src/recipes/is-date.recipe.ts [hunk 4]
```diff
@@ -1,29 +1,43 @@
...
-    }).add(
-        // TODO: validate the input string before transforming to a Date object
+            // OpenAPI expresses dates as a string format (either 'date' or 'date-time')
+            type: 'string',
+            format: options?.format || DEFAULT_FORMAT,
+        });
 
-        // convert strings to Dates
-        TypePropertyDecorator(() => Date),
+        if (options?.format === DateFormat.DATE_TIME) {
```
Comment (jessemyers) [inline src/recipes/is-date.recipe.ts]: The builder structure will filter out undefined inputs; I use this elsewhere to keep conditionals out of these recipes; you just use a ternary condition to either add a decorator or add `undefined`.

**Review by jessemyers — APPROVED** [review]

File: src/flavors/__tests__/fixtures.ts [hunk 5]
```diff
@@ -46,6 +47,9 @@ export function createFixtures(
         @IsBoolean()
         requiredBooleanValue!: boolean;
 
+        @IsDateString()
```
Comment (jessemyers) [inline src/flavors/__tests__/fixtures.ts:50]: 👍

**Review by jessemyers — APPROVED** [review]

**Pull request merged**